server/autolinkplugin/plugin.go

```diff
@@ -59,22 +58,55 @@ func (p *Plugin) IsAuthorizedAdmin(userId string) (bool, error) {
 	return false, nil
 }
 
-func contains(team string, channel string, list []string) bool {
-	for _, channelTeam := range list {
-		channelTeamSplit := strings.Split(channelTeam, "/")
-		if len(channelTeamSplit) == 2 {
-			if strings.EqualFold(channelTeamSplit[0], team) && strings.EqualFold(channelTeamSplit[1], channel) {
-				return true
-			}
-		} else if len(channelTeamSplit) == 1 {
-			if strings.EqualFold(channelTeamSplit[0], team) {
-				return true
-			}
-		} else {
-			mlog.Error("error splitting channel & team combination.")
+func (p *Plugin) resolveScope(channelId string) (string, string, *model.AppError) {
+	channel, cErr := p.API.GetChannel(channelId)
+	if cErr != nil {
+		p.API.LogError("Failed to get Channel", "error", cErr.Error())
```

Comment:
we are logging twice, here and outside of the function call. I don't think this log is necessary. You can just wrap the error and return.